pyproject.toml

```diff
@@ -150,8 +150,12 @@ exclude = [
 ]
 
 [tool.pytest.ini_options]
-python_files = "test_*.py"
-testpaths = "scanpy/tests/"
+addopts = [
+    "--import-mode=importlib",
+    "-pscanpy.testing._pytest",
+]
+norecursedirs = ["docs"]
+xfail_strict = true
```

Comment:
Could you explain these changes?

Reply:
- `"--import-mode=importlib"` makes sure we don’t mix `pytest` idioms with imports from test directories. That gets ugly. It‘s recommended, we use it in our template, …
- `"-pscanpy.testing._pytest"` as described in https://github.com/scverse/scanpy/pull/2235#discussion_r1234145378, this module is a pytest plugin and contains pytest customizations we use. It replaces the use of a top-level `conftest`
- `norecursedirs = ["docs"]` is the inversion of `testpaths`: blocklist instead of allowlist. I’m happy to change that one to `testpaths = ['scanpy']`
- `xfail_strict = true` once tests we expect to fail suddenly work, we need to check why.